core/chains/evm/txm/storage/inmemory_store.go

```diff
@@ -0,0 +1,333 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txm/types"
+)
+
+const (
+	maxQueuedTransactions = 250
```

Comment:
I noticed we also use this to limit the confirmed map's size too. Could we update the name to something more generic for both scenarios?

Reply:
Or maybe just a description/comment?

Reply:
Sort of related. Instead of limiting based on the number of confirmed transactions, did you also consider reaping based on time elapsed (sort of mimicking reaper behavior). We would probably get more variable memory usage but it would also help retain transactions for a deterministic amount of time for chains that may have higher throughput.

Reply:
Size sets a clear boundary on memory usage, so we'd have to build something that combines both. Time limit is a bit more complex for this stage. I know you have a specific use case in mind, so I'll leave it for a future iteration.